DependencyInjection/Compiler/CommandHandlerPass.php

```diff
@@ -19,25 +21,100 @@ class CommandHandlerPass implements CompilerPassInterface
      */
     public function process(ContainerBuilder $container)
     {
-        if (!$container->has('tactician.handler.locator.symfony')) {
-            throw new \Exception('Missing tactician.handler.locator.symfony definition');
-        }
-
-        $handlerLocator = $container->findDefinition('tactician.handler.locator.symfony');
-
-        $mapping = [];
+        $defaultBusId = $this->getDefaultBusId($container);
+        $busIdToHandlerMapping = [];
 
         foreach ($container->findTaggedServiceIds('tactician.handler') as $id => $tags) {
+
             foreach ($tags as $attributes) {
                 if (!isset($attributes['command'])) {
                     throw new \Exception('The tactician.handler tag must always have a command attribute');
                 }
 
-                $mapping[$attributes['command']] = $id;
+                $busId = array_key_exists('bus', $attributes) ? $attributes['bus'] : $defaultBusId;
+
+                $this->abortIfInvalidBusId($busId, $container);
+
+                $busIdToHandlerMapping[$busId][$attributes['command']] = $id;
```

Comment:
Ah, check, @boekkooi reminds me this isn't auto attached (though we could potentially do it with this change but for now, it doesn't need to be done). NEVER MIND :grin: